### ISSUE TITLE
xLearn: prediction segfaults

### ISSUE THREAD

**Issue by atkm** [atkm/avazu-ctr#13]
The xlearn model trained on 'small' segfaults when predicting on the full test set. The issue does not occur on the 'small' test set.

The test set contains features not seen in the training set, as the way the test set is converted to the ffm format is wrong.

**Comment by atkm** [comment]
The current version of xlearn makes predictions without fields with unseen features. See https://github.com/aksnzhy/xlearn/issues/50

Call the compiled binary like this: `~/code/xlearn/build/xlearn_predict ffm-data/test_site_full.ffm xlearn-out/ffm-site.out --sigmoid`
It would generate `test_site_full.ffm.out` in `ffm-data/`.
